renormalizer/mps/mp.py

```diff
@@ -913,10 +886,12 @@ def _array2mt(self, array, idx):
             mt = array.astype(self.dtype)
         else:
             mt = Matrix(array, dtype=self.dtype)
+        if mt.pdim[0] != self.pbond_list[idx]:
+            raise ValueError("Matrix physical bond dimension does not match system information")
+        mt.sigmaqn = self._get_sigmaqn(idx)
         if self.use_dummy_qn:
-            mt.sigmaqn = np.zeros(mt.pdim_prod, dtype=np.int)
-        else:
-            mt.sigmaqn = self._get_sigmaqn(idx)
+            mt.sigmaqn = np.zeros_like(mt.sigmaqn)
```

Comment:
Is `use_dummy_qn` necessary or not now? we could set sigmaqn=0 to achieve the same purpose, right?

Reply:
I'm not sure about this. I'll try to remove this to see if it is necessary or not.

Reply:
I think it would be more consistent to use sigmaqn=0 if not considering the quantum number.